### PR TITLE
Consume IBE hash function from the suite

### PR DESCRIPTION
**Pull request by CluEleSsUK**
Consume IBE hash function from the suite rather than manually using blake2s XOF

### CODE REVIEW

**Comment by CluEleSsUK** [comment]
I noticed the default hash function is sha256 and some of the tests now fail - are these still relevant if not using blake?

**Comment by nikkolasg** [comment]
Now you must restrict input msg and ciphertext to be of max size "hash.Size()"

File: encrypt/ibe/ibe.go [hunk 1]
```diff
@@ -143,38 +138,46 @@ func h3(s pairing.Suite, sigma, msg []byte) (kyber.Scalar, error) {
...
 
-func gtToHash(gt kyber.Point, length int, dst []byte) ([]byte, error) {
-	xof, err := blake2s.NewXOF(maxSize, nil)
-	if err != nil {
-		return nil, err
-	}
-	if _, err := xof.Write(dst); err != nil {
+func gtToHash(s pairing.Suite, gt kyber.Point, length int, dst []byte) ([]byte, error) {
+	hash := s.Hash()
+	hash.Reset()
```
Comment (nikkolasg) [inline encrypt/ibe/ibe.go:169]: Same for reset here.

File: share/dkg/status.go [hunk 2]
```diff
@@ -12,7 +13,10 @@ const (
 )
 
 type BitSet map[uint32]bool
-type StatusMatrix map[uint32]BitSet
+type StatusMatrix struct {
```
Comment (CluEleSsUK) [inline share/dkg/status.go:16]: Interesting, this was where I found the issue before when running with `-race` but some basic locking caused deadlock for me

Comment (AnomalRoil) [reply share/dkg/status.go:16]: Maybe you added locks to both `AllTrue` and `CompleteSuccess`? Because that would deadlock then, I think.

**Pull request merged**